libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1667,15 +1667,44 @@ string YulUtilFunctions::allocateAndInitializeMemoryStructFunction(StructType co
 		templ("allocSize", _type.memoryDataSize().str());
 
 		vector<map<string, string>> memberParams(members.size());
+		vector<string> args(members.size());
 		for (size_t i = 0; i < members.size(); ++i)
 		{
+			args[i] = "arg" + to_string(i);
 			solAssert(members[i]->memoryHeadSize() == 32, "");
-			memberParams[i]["zeroValue"] = zeroValueFunction(
+			memberParams[i]["arg"] = args[i];
+		}
+		templ("member", memberParams);
+		templ("args", joinHumanReadable(args));
+
+		return templ.render();
+	});
+}
+
+string YulUtilFunctions::allocateAndInitializeMemoryStructFunctionZero(StructType const& _type)
+{
+	string functionName = "allocate_and_initialize_memory_struct_zero" + _type.identifier();
+	return m_functionCollector.createFunction(functionName, [&]() {
+		Whiskers templ(R"(
+		function <functionName>() -> memPtr {
+			memPtr := <allocStruct>(<args>)
```

Comment:
Why don't you make this one call the other?

Reply:
But that's exactly what this line is doing?